libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -83,35 +90,58 @@
 
-		// Getting login options from intent's extras
-		LoginOptions loginOptions = LoginOptions.fromBundle(getIntent().getExtras());
+        // Getting login options from intent's extras
+        final LoginOptions loginOptions = LoginOptions.fromBundle(getIntent().getExtras());
 
-		// Protect against screenshots
-		getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE,
-				WindowManager.LayoutParams.FLAG_SECURE);
+        // Protect against screenshots
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE,
+                WindowManager.LayoutParams.FLAG_SECURE);
 
-		// Setup content view
-		setContentView(salesforceR.layoutLogin());
+        // Setup content view
+        setContentView(salesforceR.layoutLogin());
 
-		// Setup the WebView.
-		final WebView webView = (WebView) findViewById(salesforceR.idLoginWebView());
-		final WebSettings webSettings = webView.getSettings();
-		webSettings.setJavaScriptEnabled(true);
-		webSettings.setAllowFileAccessFromFileURLs(true);
-		webSettings.setJavaScriptCanOpenWindowsAutomatically(true);
-		webSettings.setDatabaseEnabled(true);
-		webSettings.setDomStorageEnabled(true);
-		EventsObservable.get().notifyEvent(EventType.AuthWebViewCreateComplete, webView);
-		webviewHelper = getOAuthWebviewHelper(this, loginOptions, webView, savedInstanceState);
+        // Setup the WebView.
+        final WebView webView = (WebView) findViewById(salesforceR.idLoginWebView());
+        final WebSettings webSettings = webView.getSettings();
+        webSettings.setJavaScriptEnabled(true);
+        webSettings.setAllowFileAccessFromFileURLs(true);
+        webSettings.setJavaScriptCanOpenWindowsAutomatically(true);
+        webSettings.setDatabaseEnabled(true);
+        webSettings.setDomStorageEnabled(true);
+        EventsObservable.get().notifyEvent(EventType.AuthWebViewCreateComplete, webView);
+        webviewHelper = getOAuthWebviewHelper(this, loginOptions, webView, savedInstanceState);
 
-		// Let observers know
-		EventsObservable.get().notifyEvent(EventType.LoginActivityCreateComplete, this);
+        // Let observers know
+        EventsObservable.get().notifyEvent(EventType.LoginActivityCreateComplete, this);
         certAuthOrLogin();
+        if (!receiverRegistered) {
+            chromeEventReceiver = new ChromeEventReceiver();
+            final IntentFilter changeServerFilter = new IntentFilter(ServerPickerActivity.CHANGE_SERVER_INTENT);
+            registerReceiver(chromeEventReceiver, changeServerFilter);
+            receiverRegistered = true;
+        }
 	}
 
+	@Override
+	protected void onDestroy() {
+        if (receiverRegistered) {
+            unregisterReceiver(chromeEventReceiver);
+            receiverRegistered = false;
+        }
+        super.onDestroy();
+    }
+
+	@Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        if (isChromeCallback(intent)) {
```

Comment:
This gets triggered only for an `Activity` that has `singleInstance` enabled in the `Manifest` and helps avoid creating the `Activity` over and over again. Many other benefits too.